File: pyg_manager_web/src/main/webapp/js/controller/brandController.js
//表现层， 和页面数据交互
app.controller("brandController", function ($scope, brandService, $controller) {
    $controller("baseController", {$scope: $scope});

    $scope.findAll = function () {
        brandService.findAll().success(function (response) {
            $scope.list = response;
        })
    };
    //保存方法
    $scope.save = function () {
        var obj = null;
        if ($scope.entity.id != null) {
            obj = brandService.update($scope.entity);
        } else {
            obj = brandService.add($scope.entity);
        }
        obj.success(function (response) {
            if (response.success) {
                $scope.reloadList();
            } else {
                alert(response.message);
            }
        });
    };
    //根据ID查询用户
    $scope.findOne = function (id) {
        brandService.findOne(id).success(function (response) {
            $scope.entity = response;
        });

    };


    //删除
    $scope.dele = function () {
        //判断数组中是否有值
        if ($scope.selectIds.length === 0) {
            alert("请选择您要删除的数据！");
            return;
        }
        var flag = window.confirm("确定要删除您选的数据么？");
        if (flag) {
            brandService.dele($scope.selectIds).success(function (response) {
                if (response.success) {
                    $scope.reloadList();
                    $scope.selectIds = {};//清空数组
                } else {
                    alert(response.message)
                }
            })
        }
    };


    $scope.searchEntity = {};
    $scope.search = function (pageNo, pageSize) {
        //$scope searchEntity   页码  条数
        brandService.search(pageNo, pageSize, $scope.searchEntity).success(function (response) {
            //response 分页后的结果  当前页数数据 list 总条数{total：100 ，rows:{}{}}
            $scope.list = response.rows;
            $scope.paginationConf.totalItems = response.total;
        })
    }

});

File: pyg_manager_web/src/main/java/com/pyg/manager/controller/BrandController.java
package com.pyg.manager.controller;

import bean.Result;
import bean.PageResult;
import com.alibaba.dubbo.config.annotation.Reference;
import com.pyg.pojo.TbBrand;
import com.pyg.sellergoods.service.BrandService;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

/**
 * @author dibulidubulijiojio
 * @version v1.0
 * @date 2019/3/8  10:31
 * @description TODO
 **/
@RestController
@RequestMapping("/brand")
public class BrandController {
    @Reference  //  远程注入  不同项目下
    private BrandService brandService;


    @RequestMapping("/findBrandList")
    public List<Map> findBrandList() {
        return brandService.findBrandList();
    }


    @RequestMapping("/findAll")
    public List<TbBrand> findAll() {
        return brandService.findAll();
    }

    @RequestMapping("findPage")
    public PageResult findPage(Integer pageNo, Integer pageSize) {
        // {total:100,rows:[{},{}]}
        return brandService.findPage(pageNo, pageSize);
    }

    @RequestMapping("add")
    public Result add(@RequestBody TbBrand tbBrand) {
        // response ={success:true|false,message:"添加成功"|"添加失败"}
        try {
            brandService.add(tbBrand);
            return new Result(true, null);
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "添加失败");
        }
    }


    @RequestMapping("findOne")
    public TbBrand findOne(Long id) {
        return brandService.findOne(id);

    }

    @RequestMapping("update")
    public Result update(@RequestBody TbBrand tbBrand) {
        try {
            brandService.update(tbBrand);
            return new Result(true, "修改成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "添加失败");
        }

    }


    @RequestMapping("dele")
    public  Result dale( Long[] ids){
        try {
            brandService.deleteByIds(ids);
            return  new Result( true,"删除成功");
        } catch (Exception e) {
            e.printStackTrace();
            return  new Result(false,"删除失败");
        }
    }

    @RequestMapping("search")
    public  PageResult search(Integer pageNo,Integer pageSize,@RequestBody TbBrand tbBrand){
       // {total:100,rows:[{},{}] }
        return brandService.findPage( pageNo,pageSize,tbBrand);
    }
}
